modules/lvgl/input/lvgl_common_input.c

```diff
@@ -52,3 +55,21 @@ int lvgl_input_register_driver(lv_indev_type_t indev_type, const struct device *
 
 	return 0;
 }
+
+#define LV_DEV_INIT(node_id, init_fn, ret) ret |= init_fn(DEVICE_DT_GET(node_id));
```

Comment:
not a big fan of trying to go forward with initialization with partial failures, if this errors out there's probably something real bad going on.

I would do this as
```c
#define LV_DEV_INIT(node_id, init_fn) do { \
        int ret = init_fn(DEVICE_DT_GET(node_id)); \
        if (ret) { \
                return ret; \
        } while (0)
```

(like https://github.com/zephyrproject-rtos/zephyr/blob/f007f21b3ce2b85f941e8b5d1fc6386ee7fc45d3/lib/os/cbprintf_complete.c#L1366C24-L1373, I think that the do loop is not strictly necessary with GCC but I'm not quite lucky with compiler quirks these days...)

Then `lvgl_init_input_devices` would just call the foreach, use the non VARGS version and `return 0` at the end.

Note that with @gmarull https://github.com/zephyrproject-rtos/zephyr/pull/61986 this would become super simple since it moves the callback in the struct device, but for now I think this is good, we can revisit later.

Reply:
yikes... yeah maybe let's go with iterable sections as Gerard is suggesting on the other comment, may be the cleanest solution.

Sorry for kicking this around so much.

Reply:
well guys, let's not overengineer this, just throwing some ideas :-) I'm fine with any reasonable solution, as soon as it does not insert soft dependencies in the init chain.

Reply:
Grateful for your help in hashing this one out :^)